src/main/java/com/zipcodewilmington/looplabs/IntegerDuplicateDeleter.java

```diff
@@ -1,8 +1,95 @@
 package com.zipcodewilmington.looplabs;
 
+import java.util.Arrays;
+
 /**
  * Created by leon on 1/29/18.
+ *
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class IntegerDuplicateDeleter extends DuplicateDeleter<Integer> {
+
+
+    public IntegerDuplicateDeleter(Integer[] intArray) {
+        super(intArray);
+    }
+
+    public Integer[] removeDuplicates(int maxNumberOfDuplications) {
+        Integer[] outputArray = this.array;
+        Integer[] toDelete = new Integer[0];
+
+        int deleteCounter = 0;
+        for (Integer x : this.array) {
+            if (valueOccurs(toDelete, x) == true) {
+                continue;
+
+            } else if (timesValueOccurs(this.array, x) >= maxNumberOfDuplications) {
+                toDelete = Arrays.copyOf(toDelete, toDelete.length + 1);
+                toDelete[deleteCounter] = x;
+                deleteCounter++;
+            }
+
+            for (Integer y : toDelete) {
+                outputArray = remove(outputArray, y);
+            }
+        }
+        System.out.println(outputArray);
+        return outputArray;
+    }
+
+    //@Override
+    public Integer[] removeDuplicatesExactly(int exactNumberOfDuplications) {
+        Integer[] outputArray = this.array;
+        Integer[] toDelete = new Integer[0];
+        int deleteCounter = 0;
+        for (Integer i : this.array) {
+            if (valueOccurs(toDelete, i) == true) {
+                continue;
```

Comment:
something seems off about the names here. I don't quite get why you continue if a value occurs. On top of that you can simply use 
`if (valueOccurs(toDelete, i)` without the `== true`. by default an if statement is looking for a true statement.

Reply:
ok I get whats going on now. you're checking if its a value that has already been set to be deleted. this block is just a bit confusing. Need better names for variables.